File: c/10828.c
#include <stdio.h>
#include <string.h>

int stack[10001] = {0, };

int main() {
  char order[6];
  int n, x, y;
  x = 0; y = 0;
  scanf("%d", &n);
  for(int i=0; i<n; i++) {
    scanf("%s", order);
    if(!strcmp(order, "push")) {
      scanf("%d", &stack[y]);
      y++;
    } else if(!strcmp(order, "pop")) {
      if(x==y) {
        printf("-1\n");
      } else {
        y--;
        printf("%d\n", stack[y]);
      }
    } else if(!strcmp(order, "size")) {
      printf("%d\n", y-x);
    } else if(!strcmp(order, "empty")) {
      if(x==y) {
        printf("1\n");
      } else {
        printf("0\n");
      }
    } else if(!strcmp(order, "top")) {
      if(x==y) {
        printf("-1\n");
      } else {
        printf("%d\n", stack[y-1]);
      }
    }
  }
  return 0;
}

File: python/10828.py
n = int(input())
stack = []
for i in range(n):
  order = input()
  if "push" in order:
    if len(stack)==0:
      stack.append(order[5])
    else:
      stack.insert(0, order[5])
  elif "pop" in order:
    if len(stack)==0:
      print(-1)
    else: 
      print(stack[0])
      del stack[0]
  elif "size" in order:
    print(len(stack))
  elif "empty" in order:
    if len(stack)==0:
      print(1)
    else:
      print(0)
  elif "top" in order:
    if len(stack)==0:
      print(-1)
    else:
      print(stack[0])

